### PR TITLE
Add support for parsing redis:// and rediss:// URLs

### PR DESCRIPTION
**Pull request by freeformz**
This includes setting up a default dialer that handles the ssl
handshake.

We (Heroku) would like to move secure redis connections forward, but
have this chicken and egg problem with some drivers. Drivers for other
languages already support rediss://

### CODE REVIEW

File: options.go [hunk 1]
```diff
@@ -92,6 +100,70 @@ func (opt *Options) init() {
...
+			conn, err := net.DialTimeout(o.Network, o.Addr, o.DialTimeout)
+			if err != nil {
+				return nil, err
+			}
+			if o.TLSConfig == nil {
+				o.TLSConfig = &tls.Config{InsecureSkipVerify: true}
+			}
+			t := tls.Client(conn, o.TLSConfig)
+			return t, t.Handshake()
+		}
```
Comment (vmihailenco) [inline options.go]: I think this should be moved to https://github.com/go-redis/redis/blob/v5/options.go#L70

File: options.go [hunk 2]
```diff
@@ -92,6 +100,70 @@ func (opt *Options) init() {
...
+		h = "localhost"
+	}
+	if p == "" {
+		p = "6379"
+	}
+	o.Addr = net.JoinHostPort(h, p)
+
+	f := strings.FieldsFunc(u.Path, func(r rune) bool {
+		return r == '/'
+	})
```
Comment (vmihailenco) [inline options.go]: Is not this the same as `strings.Split(u.Path, "/")`? Just asking :)

File: options_test.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+			"",
+			0, false, errors.New("Invalid redis database number: strconv.ParseInt: parsing \"iamadatabase\": invalid syntax"),
+		},
+	}
+
+	for _, c := range cases {
+		t.Run(c.u, func(t *testing.T) {
+			o, err := ParseURL(c.u)
+			if c.err == nil && err != nil {
+				t.Fatalf("Expected err to be nil, but got: '%q'", err)
```
Comment (vmihailenco) [inline options_test.go]: nitpics:
- start with lower case
- got goes first, want (expected) goes second
- no need to quote `%q` (at least I don't see why)

For example:
- unexpected error: %q
- got %q, want %q

**Comment by vmihailenco** [comment]
This looks good - thanks! I've left some minor comments, but they are not blockers.

I wonder if it is possible to connect to Redis Cluster using `redis://` URLs? I spent few minutes on it but can't google anything...

File: options.go (hunk 2 above)
Comment (freeformz) [inline options.go]: Nope https://play.golang.org/p/OAOF51zR-a

**Comment by freeformz** [comment]
@vmihailenco Pushed changes to address feedback.

**Comment by freeformz** [comment]
WRT redis cluster, we could address that in another PR as well.

**Comment by vmihailenco** [comment]
Looks good - thanks

**Pull request merged**